File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        map<int, int>m;
        int size = nums.size();
        for (int i = 0; i < size; i++)
        {
            auto it = m.find(nums[i]);
            if (it != m.end())
            {
                if (abs(i - it->second) <= k)
                    return (true);
            }
            m[nums[i]] = i;
        }
        return (false);
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        my_map = {};
        for index, value in enumerate(nums):
            if value in my_map:
                if (abs(index - my_map[value]) <= k):
                    return (True);
                else:
                    my_map[value] = index;
            else:
                my_map[value] = index;
        return (False);
            